pkgs/development/python-modules/python-glanceclient/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib, buildPythonApplication, fetchPypi
+, pbr , prettytable, keystoneauth1, requests, warlock, oslo-utils, oslo-i18n, wrapt, pyopenssl
+, stestr, testscenarios, ddt, requests-mock, coreutils
+}:
+
+buildPythonApplication rec {
+  pname = "python-glanceclient";
+  version = "3.5.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "417b9d814b43e62df4351f26a0d5569b801e9f99f7758bd8c82ef994c3629356";
+  };
+
+  buildNativeInputs = [
+    pbr
+  ];
+
+  propagatedBuildInputs = [
+    prettytable
+    keystoneauth1
+    requests
+    warlock
+    oslo-utils
+    oslo-i18n
+    wrapt
+    pyopenssl
+  ];
+
+  checkInputs = [
+    stestr
+    testscenarios
+    ddt
+    requests-mock
+  ];
+
+  checkPhase = ''
+    substituteInPlace glanceclient/tests/unit/v1/test_shell.py \
+      --replace "/bin/echo" "${coreutils}/bin/echo"
+
+    stestr run
+
+    $out/bin/glance --version | grep ${version} > /dev/null
```

Comment:
I don't think it is worth to add those to all tools. Missing clients are only detected at runtime and the import check should be almost equivalent if good practices are used upstream and is enforced by tooling.